applications/vanilla/modules/class.tagmodule.php

```diff
@@ -102,6 +102,16 @@ public function getData() {
         $tagQuery = Gdn::sql();
 
         $this->autoContext();
+        //Allow addon to manipulate the data being rendered
+        $tagData = null;
+        $this->EventArguments['ParentID'] = $this->ParentID;
+        $this->EventArguments['tagData'] = &$tagData;
+        $this->fireEvent('getData');
+        if ($tagData !== null) {
```

Comment:
The expectation is that, after this method is run, *something* will be in the `_TagData` property. The logic here doesn't look like it will allow that. First, we're checking to see if it's `null`. That's good. We're attempting to confirm an addon modified the value.  Next, we're making sure the value is an array. If it is, we're updating the `_TagData` property with the value and exiting the method. If it isn't an array, we're just exiting the method. `_TagData` is never updated with anything.

Reply:
That definitely wasn't the case [when I commented on it](https://github.com/vanilla/vanilla/blob/22718c1bba70a1df954bea0959a42890ed3a039f/applications/vanilla/modules/class.tagmodule.php#L110).